http-client/src/main/java/io/airlift/http/client/spnego/SpnegoAuthentication.java

```diff
@@ -193,6 +161,58 @@ public boolean matches(String type, URI uri, String realm)
         return NEGOTIATE.equalsIgnoreCase(type);
     }
 
+    private void ensureClientCredentialUpdated()
+            throws LoginException, GSSException
+    {
+        if (clientCredential == null || clientCredential.getRemainingLifetime() <= MIN_CREDENTIAL_LIFE_TIME_SEC) {
+            if (loginContext != null) {
+                loginContext.logout();
```

Comment:
I believe we don't need the treatment for SpnegoFilter. The server credential creation does not go through KDC, it's directly derived from the keytab which does not have expiration time.